torch/_inductor/codegen/cuda/cuda_scheduling.py

```diff
@@ -1,10 +1,80 @@
+from typing import cast, List
+
 from ... import config
 from ...codecache import code_hash, get_path
+from ...scheduler import FusedSchedulerNode, Scheduler, SchedulerNode
 from ...utils import get_fused_kernel_name, get_kernel_metadata
 from ...virtualized import V
-
 from ..common import IndentedBuffer
 from ..triton import TritonScheduling
+from .cuda_kernel import CUDATemplateBuffer
+
+
+class CUDASchedulerNode(SchedulerNode):
```

Comment:
SchedulerNode which wraps a CUDATemplateBuffer. If those two classes wouldn't need to have different baseclasses, we might have wanted to merge them. But they belong to different API layers. This one belongs to the Scheduler, the CUDATemplateBuffer is an IRNode / ir.Buffer